pop/service/service.go

```diff
@@ -224,47 +466,451 @@ func (s *Service) CheckConfig(req *network.Envelope) {
 // CheckConfigReply strips the attendees missing in the reply, if the
 // PopStatus == PopStatusOK.
 func (s *Service) CheckConfigReply(req *network.Envelope) {
-	ccrVal, ok := req.Msg.(*CheckConfigReply)
-	var ccr *CheckConfigReply
-	ccr = func() *CheckConfigReply {
+	ccrVal, ok := req.Msg.(*checkConfigReply)
+	var ccr *checkConfigReply
+	ccr = func() *checkConfigReply {
 		if !ok {
 			log.Errorf("Didn't get a CheckConfigReply: %v", req.Msg)
 			return nil
 		}
-		if !bytes.Equal(ccrVal.PopHash, s.data.Final.Desc.Hash()) {
-			log.Error("Not correct hash")
+		var final *FinalStatement
+		if final, ok = s.data.Finals[string(ccrVal.PopHash)]; !ok {
+			log.Error("No party with given hash")
 			return nil
 		}
 		if ccrVal.PopStatus < PopStatusOK {
-			log.Lvl1("Wrong pop-status:", ccrVal.PopStatus)
+			log.Error("Wrong pop-status:", ccrVal.PopStatus)
 			return nil
 		}
-		s.intersectAttendees(ccrVal.Attendees)
+		final.Attendees = intersectAttendees(final.Attendees, ccrVal.Attendees)
 		return ccrVal
 	}()
-	if len(s.ccChannel) == 0 {
-		s.ccChannel <- ccr
+	if syncData, ok := s.syncs[string(ccrVal.PopHash)]; ok {
+		if len(syncData.ccChannel) == 0 {
+			syncData.ccChannel <- ccr
```

Comment:
same as above: as `ccr` is not used outside of the if, call the function only here.